include/NovelRT.Interop/Graphics/NrtRenderingService.h

```diff
@@ -1,35 +1,52 @@
-  NrtResult Nrt_RenderingService_initialiseRendering(NrtRenderingService renderingService); //this function has its own return codes, so in theory we can combine them both into the main return code
-  NrtResult Nrt_RenderingService_tearDown(NrtRenderingService renderingService);
-  NrtResult Nrt_RenderingService_createImageRectWithFile(NrtRenderingService renderingService, NrtImageRect* outputImageRect, NrtTransform transform, int32_t layer, const char* filePath, NrtRGBAConfig colourTint);
-  NrtResult Nrt_RenderingService_createImageRectWithNothing(NrtRenderingService renderingService, NrtImageRect* outputImageRect, NrtTransform transform, int32_t layer, NrtRGBAConfig colourTint);
-  NrtResult Nrt_RenderingService_createBasicFillRect(NrtRenderingService renderingService, NrtBasicFillRect* outputBasicFillRect, NrtTransform transform, int32_t layer, NrtRGBAConfig colourConfig);
-  NrtResult Nrt_RenderingService_createTextRect(NrtRenderingService renderingService, NrtTextRect* outputTextRect, NrtTransform transform, int32_t layer, NrtRGBAConfig colourConfig, float fontSize, const char* fontFilePath);
-  NrtResult Nrt_RenderingService_getCamera(NrtRenderingService renderingService, NrtCamera* outputCamera);
-  NrtResult Nrt_RenderingService_beginFrame(NrtRenderingService renderingService);
-  NrtResult Nrt_RenderingService_endFrame(NrtRenderingService renderingService);
-  NrtResult Nrt_RenderingService_setBackgroundColour(NrtRenderingService renderingService, NrtRGBAConfig colour);
-  NrtResult Nrt_RenderingService_getTextureWithNothing(NrtRenderingService renderingService, NrtTexture* outputTexture);
-  NrtResult Nrt_RenderingService_getTextureWithFile(NrtRenderingService renderingService, NrtTexture* outputTexture, const char* fileTarget);
-  NrtResult Nrt_RenderingService_getFontSet(NrtRenderingService renderingService, NrtFontSet* outputFontSet, const char* fileTarget, float fontSize, const char**errorMessage);
-  NrtResult Nrt_RenderingService_destroy(NrtRenderingService renderingService);
+    NrtResult Nrt_RenderingService_create(NrtWindowingService windowingService,
+                                          NrtRenderingService* outputRenderingService);
+    NrtResult Nrt_RenderingService_initialiseRendering(
+        NrtRenderingService renderingService); // this function has its own return codes, so in theory we can combine
+                                               // them both into the main return code
+    NrtResult Nrt_RenderingService_tearDown(NrtRenderingService renderingService);
+    NrtResult Nrt_RenderingService_createImageRectWithFile(NrtRenderingService renderingService,
+                                                           NrtImageRect* outputImageRect, NrtTransform transform,
+                                                           int32_t layer, const char* filePath,
+                                                           NrtRGBAConfig colourTint);
```

Comment:
Is this how we want to handle method parameters? It reads a bit odd to me but would appreciate further input.

Reply:
I can attempt to binpack the params or allow them to each be on their own line if it won't fit on one, however this will probably cause some of our more param heavy methods to expand a bit space-wise.

@FiniteReality / @tannergooding any thoughts?

Reply:
I think I'd prefer not to hav emultiple on the same line when they are split like this, it makes things like `transform` and `filePath` easy to miss

Reply:
For C/C++, I personally prefer this:
```
void MyLongMethod(
    LongParameter1 myLongParameter,
    LongParameter2 myOtherLongParameter,
    LongParameter3 andSoOn);

Reply:
Should be resolved - I've split all parameters onto separate lines.